### DISCUSSION TITLE
Showing Automation Progress in the Front End

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7697] [Help] [answered]
I am looking for a way to show to the user the progress of the Automation he fired.
I have an automation, that sends batches of SMS messages, say 5000.

So I have a TextBox and a Button, when the user hits the button, the Automation will be triggered and it will auto send the contents of the textbox to a list of people.

What I would like to achieve, is to have a counter in the screen showing progress as in "Sending Message Number #".

Is there a way to send info from the Automation to the Screen ? 

I could add an update row action in the Automation to keep the count of the position, but that would be just a big waste of resources. Is something like "Update State" from within an Automation possible ?

**Comment by melohagan** [comment, marked as answer]
Hey @poirazis 

This is not possible because automations are run via the worker container asynchronously. There is no concept of the front-end from the automations' point of view. 

There is also currently no websocket or pub/sub functionality within Budibase that would allow for live updates.
